src/components/LearnerActivityTable/LearnerActivityTable.test.jsx

```diff
@@ -107,33 +36,54 @@ const LearnerActivityTableWrapper = props => (
 );
 
 const verifyLearnerActivityTableRendered = (tableId, activity, columnTitles, rowsData) => {
-  const wrapper = mount((
-    <LearnerActivityTableWrapper id={tableId} activity={activity} />
-  ));
-  // Verify that table has correct number of columns
-  expect(wrapper.find(`.${tableId} thead th`).length).toEqual(columnTitles.length);
+  const wrapper = mount(<LearnerActivityTableWrapper id={tableId} activity={activity} />);
+
+  const table = wrapper.find('[role="table"]');
+  const headerColumns = table.find('thead th');
+  const tableRows = table.find('tbody tr');
 
-  // Verify only expected columns are shown
-  wrapper.find(`.${tableId} thead th`).forEach((column, index) => {
+  // Verify the number of columns
+  expect(headerColumns).toHaveLength(columnTitles.length);
+
+  // Verify column titles
+  headerColumns.forEach((column, index) => {
     expect(column.text()).toContain(columnTitles[index]);
   });
 
-  // Verify that table has correct number of rows
-  expect(wrapper.find(`.${tableId} tbody tr`).length).toEqual(2);
+  // Verify the number of rows
+  expect(tableRows).toHaveLength(rowsData.length);
 
-  // Verify each row in table has correct data
-  wrapper.find(`.${tableId} tbody tr`).forEach((row, rowIndex) => {
-    row.find('td').forEach((cell, colIndex) => {
+  // Verify row data
+  tableRows.forEach((row, rowIndex) => {
+    const cells = row.find('td');
+    cells.forEach((cell, colIndex) => {
       expect(cell.text()).toEqual(rowsData[rowIndex][colIndex]);
     });
   });
 };
 
 describe('LearnerActivityTable', () => {
+  beforeEach(() => {
+    useCourseEnrollments.mockReturnValue(mockUseCourseEnrollments);
+  });
+
+  afterEach(() => jest.clearAllMocks());
+
   it('renders empty state correctly', () => {
+    useCourseEnrollments.mockReturnValue(
+      {
+        isLoading: false,
+        courseEnrollments: {
+          itemCount: 0,
+          pageCount: 0,
+          results: [],
+        },
+        fetchCourseEnrollments: jest.fn(),
+      },
+    );
     const tree = renderer
       .create((
-        <LearnerActivityEmptyTableWrapper id="active-week" activity="active_past_week" />
+        <LearnerActivityTableWrapper id="active-week" activity="active_past_week" />
```

Comment:
[suggestion] We want to try and move away from using snapshots as a form of testing. These should be refactored to no longer generate snapshots.